grunttasks/scan-email-translations.js

```diff
@@ -0,0 +1,108 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+module.exports = function (grunt) {
+  grunt.registerTask('scan-email-translations', 'Scan the generated emails for invalid style tags', [
+    'generate-valid-styles',
+    'po2json',
+    'verify-styles'
+    ]
+  );
+
+  grunt.registerMultiTask('verify-styles', 'Verify whether the translated styles are valid', function (){
+    var fileArray = this.files[0].src;
+    var contents = grunt.file.read(fileArray.shift());
+    var styleArray = JSON.parse(contents);
+
+    var hasErrors = 0;
+    var styleRegex = /style="(.*?)"/ig;
+
+    fileArray.forEach(function (src) {
+      var json = JSON.parse(grunt.file.read(src));
+      var styleList = [];
+      for(var key in json) {
+        if(key.indexOf('style') > 0) {
+          styleList.push(key);
+        }
+      }
+      // Check if there were any styles found, .match() might return null
+      if (styleList !== null) {
+        styleList = styleList.map(stripSpecialChars);
+        // retrieve the style attribute and its contents and store
+        styleList = styleList.map(function (str) {
+          if (styleRegex.test(str)) {
+            str = str.match(styleRegex)[0];
+            return str;
+          }
+        });
+        styleList.forEach(function (style) {
+          if(style !== undefined){
+            if (styleArray.indexOf(style) === -1) {
+              grunt.log.error('Found a style which should not be there', style, src);
+              hasErrors = hasErrors + 1;
+            }
+          }
+        });
+      }
+    });
+    if (hasErrors !== 0) {
+      grunt.fail.warn('Found ' + hasErrors + ' ' + grunt.util.pluralize(hasErrors, 'error/errors') + ' in ' + fileArray.length + ' files');
+    } else {
+      grunt.log.writeln('Checked ' + fileArray.length + ' files for valid styles and found nothing wrong.');
+    }
+  });
+
+  grunt.config('verify-styles', {
+    dist: {
+      files: [
+        {
+          expand: false,
+          src: [
+            '.tmp/validStyles.json',
+            '.tmp/l10n/*/messages.json'
```

Comment:
Do we want to lint the `.tmp/l10n/*/client.json` files as well, or just `messages.json`?
Also, do we need to verify the validStyles.json file since that was already generated via `generate-valid-styles` task?